darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1726,29 +1754,80 @@ def load(path: str, **kwargs) -> "TorchForecastingModel":
+            Running on GPU(s) is also possible using ``pl_trainer_kwargs`` by specifying keys ``"accelerator",
+            "devices", and "auto_select_gpus"``. Some examples for setting the devices inside the ``pl_trainer_kwargs``
+            dict:
+
+            - ``{"accelerator": "cpu"}`` for CPU,
+            - ``{"accelerator": "gpu", "devices": [i]}`` to use only GPU ``i`` (``i`` must be an integer),
+            - ``{"accelerator": "gpu", "devices": -1, "auto_select_gpus": True}`` to use all available GPUS.
+
+            For more info, see here:
+            `trainer flags
+            <https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#trainer-flags>`_,
+            and `training on multiple gpus
+            <https://pytorch-lightning.readthedocs.io/en/stable/accelerators/gpu_basic.html#train-on-multiple-gpus>`_.
+
+            With parameter ``"callbacks"`` you can add custom or PyTorch-Lightning built-in callbacks to Darts'
+            :class:`TorchForecastingModel`.
+            Note that you can also use a custom PyTorch Lightning Trainer for training and prediction with optional
+            parameter ``trainer`` in :func:`fit()` and :func:`predict()`.
         **kwargs
             Additional kwargs for PyTorch Lightning's :func:`LightningModule.load_from_checkpoint()` method,
             such as ``map_location`` to load the model onto a different device than the one from which it was saved.
             For more information, read the `official documentation <https://pytorch-lightning.readthedocs.io/en/stable/
             common/lightning_module.html#load-from-checkpoint>`_.
         """
-
         # load the base TorchForecastingModel (does not contain the actual PyTorch LightningModule)
+        map_location = kwargs.get("map_location", None)
         with open(path, "rb") as fin:
-            model: TorchForecastingModel = torch.load(
-                fin, map_location=kwargs.get("map_location", None)
-            )
+            model: TorchForecastingModel = torch.load(fin, map_location=map_location)
+
+        reset_trainer_params = model.trainer_params is None
+        if reset_trainer_params:
+            model.trainer_params = {"callbacks": []}
 
         # if a checkpoint was saved, we also load the PyTorch LightningModule from checkpoint
         path_ptl_ckpt = path + ".ckpt"
         if os.path.exists(path_ptl_ckpt):
             model.model = model._load_from_checkpoint(path_ptl_ckpt, **kwargs)
+
+            if reset_trainer_params:
+                dtype_to_precision = (
+                    {torch.float32: "32", torch.float64: "64"}
+                    if not pl_200_or_above
+                    else {torch.float32: "32-true", torch.float64: "64-true"}
+                )
+                model.trainer_params["precision"] = dtype_to_precision[
+                    model.model.dtype
+                ]
+                model.trainer_params["accelerator"] = model.model.device.type
```

Comment:
Actually, I believe the precision is already restored when loading the model checkpoint: https://github.com/unit8co/darts/blob/c48521ce9bbadbd9adde6df2dfb5797c60aad514/darts/models/forecasting/pl_forecasting_module.py#L548

In that case we could remove the code below. The accelerator they can set with the `pl_trainer_kwargs`

```suggestion
```